modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/exportimport/portlet/preferences/processor/DLExportImportPortletPreferencesProcessor.java

```diff
@@ -530,50 +545,204 @@ private Folder _getFolder(
+	private Folder _getFolder(
+			String folderExternalReferenceCode,
+			String groupExternalReferenceCode,
+			PortletDataContext portletDataContext)
+		throws PortletDataException {
+
+		try {
+			Group group = _groupLocalService.getGroupByExternalReferenceCode(
+				groupExternalReferenceCode, portletDataContext.getCompanyId());
+
+			DLFolder dlFolder =
+				_dlFolderLocalService.getDLFolderByExternalReferenceCode(
+					folderExternalReferenceCode, group.getGroupId());
+
+			if (!dlFolder.isInTrash()) {
+				return new LiferayFolder(dlFolder);
+			}
+		}
+		catch (PortalException portalException) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					StringBundler.concat(
+						"Portlet ", portletDataContext.getPortletId(),
+						" refers to an invalid root folder ERC ",
+						folderExternalReferenceCode, " with group ERC ",
+						groupExternalReferenceCode),
+					portalException);
+			}
+		}
+
+		return null;
+	}
+
+	private long _getFolderId(
+		Element folderElement, PortletDataContext portletDataContext) {
+
+		Folder folder = (Folder)portletDataContext.getZipEntryAsObject(
+			folderElement, folderElement.attributeValue("path"));
+
+		return folder.getFolderId();
+	}
+
+	private String _getGroupExternalReferenceCode(Folder folder) {
+		Group group = _groupLocalService.fetchGroup(folder.getGroupId());
+
+		if (group == null) {
+			return StringPool.BLANK;
+		}
+
+		return group.getExternalReferenceCode();
+	}
+
+	private String _getMirrorGroupExternalReferenceCode(
+		PortletDataContext portletDataContext,
+		String groupExternalReferenceCode) {
+
+		Group group = _groupLocalService.fetchGroupByExternalReferenceCode(
+			groupExternalReferenceCode, portletDataContext.getCompanyId());
 
 		if (group == null) {
-			return repositoryId;
+			return groupExternalReferenceCode;
 		}
 
 		Group stagingGroup = group.getStagingGroup();
 
 		if (stagingGroup != null) {
-			return stagingGroup.getGroupId();
+			return stagingGroup.getExternalReferenceCode();
+		}
+
+		if (ExportImportThreadLocal.isStagingInProcess() &&
+			group.isStagedRemotely()) {
+
+			UnicodeProperties typeSettingsUnicodeProperties =
+				group.getTypeSettingsProperties();
+
+			String remoteGroupExternalReferenceCode =
+				typeSettingsUnicodeProperties.get(
+					"remoteGroupExternalReferenceCode");
+
+			if (Validator.isNull(remoteGroupExternalReferenceCode)) {
+				remoteGroupExternalReferenceCode =
+					_getRemoteGroupExternalReferenceCode(
+						typeSettingsUnicodeProperties);
+			}
+
+			if (Validator.isNotNull(remoteGroupExternalReferenceCode)) {
+				groupExternalReferenceCode = remoteGroupExternalReferenceCode;
+			}
+		}
+
+		Group liveGroup = _groupLocalService.fetchGroup(group.getLiveGroupId());
+
+		if (liveGroup == null) {
+			return groupExternalReferenceCode;
+		}
+
+		return liveGroup.getExternalReferenceCode();
+	}
+
+	private String _getRemoteGroupExternalReferenceCode(
+		UnicodeProperties typeSettingsUnicodeProperties) {
```

Comment:
(optional; maybe for later) Wouldn't make sense to move this into `GroupImpl`? This looks like something that will be needed at a lot of other places.

Reply:
I agree that this code can be reused, however I feel it should be in some other Util class. We already have the Type Setting property "remoteGroupExternalReferenceCode":
* We could have in `GroupImpl` a method similar to `getRemoteLiveGroupId`
* The code should be used only in case of a missing property (and I would set the Type Setting by the way)

Reply:
We can discuss this later to not delay this PR.